### ISSUE TITLE
Customize `quarkus.oidc.authentication.java-script-auto-redirect` browser side header requirements.

### ISSUE THREAD

**Issue by sberyozkin** [quarkusio/quarkus#33436]
### Discussed in https://github.com/quarkusio/quarkus/discussions/33434

<div type='discussions-op-text'>

<sup>Originally posted by **kkazuo** May 17, 2023</sup>
Can I customize this `X-Requested-With: JavaScript` browser side cooporation.
https://github.com/quarkusio/quarkus/blob/f6cf2270c54aeb90e21f1b7bf159ce395b88ce0b/docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc#single-page-applications

I use the JS library that add some custom header like `HX-Request: true` automatically.
And so if quarkus-oidc can customize how to detect JS request, it's so nice.
</div>

**Comment by sberyozkin** [comment]
Hey @geoand This is the first issue which will benefit from `@Tenant("id")`, it is quite obvious I'd have come up with another pair or properties for this case :-), but here, it can really be customized only with a custom bean, possibly on a per-tenant basis, so we just do:
```
@TenantId("my-tenant")
class MyFeatures implements JavaScriptChecker, TokenCustomizer {
    
}
```

This is why I liked `Tenant("id")` approach you proposed

**Comment by geoand** [comment]
🎉

**Comment by sberyozkin** [comment]
I'll try to look shortly

**Comment by sberyozkin** [comment]
Fixed by #34812
